pkg/controller/storagecluster/storagecluster.go

```diff
@@ -643,6 +640,108 @@ func (c *Controller) syncNodes(
 	return utilerrors.NewAggregate(errors)
 }
 
+func (c *Controller) podTemplatesForNodes(
+	cluster *corev1alpha1.StorageCluster,
+	nodesNeedingStoragePods []string,
+	hash string,
+) ([]string, []*v1.PodTemplateSpec, error) {
+	nodeList := &v1.NodeList{}
+	err := c.client.List(context.TODO(), nodeList, &client.ListOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	remainingNodes := make(map[string]*v1.Node)
+	for _, nodeName := range nodesNeedingStoragePods {
+		for _, node := range nodeList.Items {
+			if node.Name == nodeName {
```

Comment:
what if there is a node present in the nodesNeedingStoragePods, but not present in nodeList.Items ?
Even though the chances of that happening are slim, we should at least log it, as we are emptying out the nodesNeedingStoragePods slice later.

Reply:
If we don't find the node in nodeList, then we should not even create a template or the pod itself as the node does not exist. The nodesNeedingStoragePods is created by us is the same reconcile loop.  The final list is only logged just before creating pods, so what we do here does not matter much as it is an intermediate step.